src/lib/concurrency/transaction_manager.cpp

```diff
@@ -13,6 +12,8 @@ void TransactionManager::reset() {
   manager._next_transaction_id = INITIAL_TRANSACTION_ID;
   manager._last_commit_id = INITIAL_COMMIT_ID;
   manager._last_commit_context = std::make_shared<CommitContext>(INITIAL_COMMIT_ID);
+  DebugAssert(manager._active_snapshot_commit_ids.empty(),
```

Comment:
As this is a fast, seldomly executed check that could potentially point to a grave problem, I'd make this an `Assert` so release builds profit from it as well.

Reply:
Good point, especially since we only call this in test code anyway